### PR TITLE
fix beacon_and_bridge spider

### PR DESCRIPTION
**Pull request by abbensid2**
This is a fix for the beacon_and_bridge spider

### CODE REVIEW

File: locations/spiders/beacon_and_bridge.py [hunk 1]
```diff
@@ -1,57 +1,83 @@
...
+                        ),
+                        "%H:%M",
+                    ),
+                    close_time=datetime.strftime(
+                        datetime.strptime(
+                            h_h[1].replace("am", " am").replace("pm", " pm"),
+                            "%I:%M %p",
+                        ),
+                        "%H:%M",
+                    ),
```
Comment (Cj-Malone) [inline locations/spiders/beacon_and_bridge.py:44]: Is there a reason to do `strftime` and `strptime`? I've not tested it, but it seems like the following should work.

```suggestion
                    open_time = h_h[0],
                    close_time = h_h[1],
                    time_format = "%I:%M%p"
```

Comment (abbensid2) [reply locations/spiders/beacon_and_bridge.py:44]: Thanks for your suggestion

File: locations/spiders/beacon_and_bridge.py [hunk 2]
```diff
@@ -1,57 +1,64 @@
...
-HEADERS = {
-    "Accept-Language": "en-US,en;q=0.9,ru;q=0.6",
-    "Origin": "http://www.beaconandbridge.com",
-    "Accept-Encoding": "gzip, deflate, br",
-    "Accept": "application/json, text/plain, */*",
-    "Referer": "http://www.beaconandbridge.com/locations/",
-    "Connection": "keep-alive",
-    "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
-}
+DAYS = ["Mo", "Tu", "We", "Th", "Fr", "Sa", "Su"]
```
[the remark this answers is not in the document]
Comment (abbensid2) [reply locations/spiders/beacon_and_bridge.py]: Thank you for your comment which made me think of recovering the data differently. at first, the loop is used to iterate through the five variables and find the data for each location cleanly.

File: locations/spiders/beacon_and_bridge.py [hunk 3]
```diff
@@ -1,57 +1,64 @@
...
+        phones = response.xpath('//div[@id="section_2"]//p/text()[1]')
+        hours = response.xpath('//div[@id="section_2"]//p/text()[2]')
+        for (
+            addr,
+            cs,
+            postcd,
+            name,
+            phone,
+            hour,
+        ) in zip(address_full, city_state, postcode, names, phones, hours):
```
[the remark this answers is not in the document]
Comment (abbensid2) [reply locations/spiders/beacon_and_bridge.py]: Thank you for your comment which made me think of recovering the data differently. at first, the loop is used to iterate through the five variables and find the data for each location cleanly.

[the remark this answers is not in the document]
Comment (Cj-Malone) [reply locations/spiders/beacon_and_bridge.py]: I think we could something like the following where we scope the following xpaths to store div, without having to have long xpaths with i in them.

```python
for store in response.xpath('//div[@id="section_2"]//div[contains(@class, "yes_heads")]'):
  item["name"] = store.xpath('.//h3[contains(@class, "the_list_item_headline")]/text()').get())
```

[the remark this answers is not in the document]
Comment (abbensid2) [reply locations/spiders/beacon_and_bridge.py]: Thanks

File: locations/spiders/beacon_and_bridge.py [hunk 4]
```diff
@@ -1,57 +1,40 @@
...
-                "addr_full": data["address"],
-                "city": data["city"],
-                "state": data["state"],
-                "postcode": data["zip"],
-                "country": data["country"],
-                "phone": data["phone"],
-                "website": data["url"],
+                "ref": re.split(" - | -", store.xpath(".//h3/text()").get())[0],
+                "name": re.split(" - | -", store.xpath(".//h3/text()").get())[1],
+                "addr_full": store.xpath(".//span/a/text()[1]").get(),
```
Comment (Cj-Malone) [inline locations/spiders/beacon_and_bridge.py:32]:
```suggestion
                "street_address": store.xpath(".//span/a/text()[1]").get(),
```

**Review by Cj-Malone — APPROVED** [review]
LGTM, thank you

**Pull request merged**